core/src/test/java/org/apache/iceberg/TestSchemaUpdate.java

```diff
@@ -488,19 +488,71 @@ public void testAmbiguousAdd() {
   @Test
   public void testAddAlreadyExists() {
     AssertHelpers.assertThrows("Should reject column name that already exists",
-        IllegalArgumentException.class, "already exists: preferences.feature1", () -> {
+        IllegalArgumentException.class, "already exists and is not being deleted: preferences.feature1", () -> {
           UpdateSchema update = new SchemaUpdate(SCHEMA, SCHEMA_LAST_COLUMN_ID);
           update.addColumn("preferences", "feature1", Types.BooleanType.get());
         }
     );
     AssertHelpers.assertThrows("Should reject column name that already exists",
-        IllegalArgumentException.class, "already exists: preferences", () -> {
+        IllegalArgumentException.class, "already exists and is not being deleted: preferences", () -> {
           UpdateSchema update = new SchemaUpdate(SCHEMA, SCHEMA_LAST_COLUMN_ID);
           update.addColumn("preferences", Types.BooleanType.get());
         }
     );
   }
 
+  @Test
+  public void testDeleteThenAdd() {
+    Schema schema = new Schema(required(1, "id", Types.IntegerType.get()));
+    Schema expected = new Schema(optional(2, "id", Types.IntegerType.get()));
+
+    Schema updated = new SchemaUpdate(schema, 1)
+            .deleteColumn("id")
```

Comment:
Nit: indentation is off in this addition.

Reply:
@thomaschow, this is still not correct. Can you fix it?

Reply:
sorry, should be fixed now I think.